feat(titleService): set pageTitle via state objects

timkindberg opened the pull request:
See https://github.com/joshdmiller/ng-boilerplate/issues/55

timkindberg committed:
#55 Set pageTitle via state object

timkindberg committed:
#55 Set pageTitle via state object

timkindberg committed:
Merge remote-tracking branch 'origin/master'

ProLoser commented on .gitignore:
```diff
@@ -4,3 +4,4 @@ build/
 bin/
 node_modules/
 vendor/
+.idea/
```
People really need to start using a global .gitignore instead of cluttering projects with their own IDE setups.

timkindberg committed:
removed the junk

timkindberg committed:
Removed titleService service, opting for scope variable instead
- Deleted titleService files and folder and all traces of it in READMEs
- Setting 'pageTitle', a AppCtrl scope variable instead, which is bound to the title tag using ng-bind
- Put the code back into AppCtrl controller, it now makes more sense there since <title> falls under AppCtrl's domain
- Fixed up several README inconsistencies

timkindberg committed:
Removed titleService service, opting for scope variable instead
- Deleted titleService files and folder
- Setting 'pageTitle', a AppCtrl scope variable instead, which is bound to the title tag using ng-bind
- Put the code back into AppCtrl controller, it now makes more sense there since <title> falls under AppCtrl's domain

timkindberg committed:
/#55 set pageTitle via state object
- Removed titleService service, opting for scope variable instead
- Deleted titleService files and folder
- Setting 'pageTitle', a AppCtrl scope variable instead, which is bound to the title tag using ng-bind
- Put the code back into AppCtrl controller, it now makes more sense there since <title> falls under AppCtrl's domain

timkindberg commented:
Sorry about the messy git flubbery. I'm still learning :/. You'll have to look at both of the "Removed titleService service..." commits to get the full picture.

joshdmiller commented:
Thanks for submitting this. To make this easier to look at, can you squash the commits?

Basically, when the changes in a PR are revised (a natural part of the development process) the PR itself becomes harder to evaluate because one cannot easily (that is, from within GitHub) look at all of the changes at once; instead I have to browse through all of the individual commits and try to add them together in my head, which is error-prone. So a common tactic with PRs is to squash the commits, meaning to take all of them and throw them into one commit that's easier to evaluate.

It takes only a couple commands. Here's a little how-to: http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request

Thanks!

ProLoser commented:
@joshdmiller check out: http://ejohn.org/blog/pulley/

I want to start using this across more projects to make it easier (and less beaurocratic) for people to contribute.

timkindberg commented:
Hey @joshdmiller. I'm sorry man I attempted to squash my commits and was not successful. I'm not sure what values to use to replace the 'upstream' and 'omgpull' from that article you posted.

```
$ git fetch upstream??
$ git checkout omgpull?? 
$ git rebase upstream??/master
$ git rebase -i

< choose squash for all of your commits, except the first one >
< Edit the commit message to make sense, and describe all your changes >

$ git push origin omgpull?? -f
```

imma keep trying

timkindberg commented:
Think I may have it figured out...

joshdmiller commented:
@timkindberg The "upstream" is the name of the remote repository (if you cloned, it's "origin") and the "omgpull" is your local working branch (master).

What you're really trying to do is this: you want to tell your repository to rewind back to the point where you diverged from ngBoilerplate (the upstream/origin) and then have the repository "replay" all your changes on top of it without committing; so the result is that your file tree looks the same, but those old commits are gone (but their changes aren't!).

An d then you need to commit your changes with one commit (squashing all the rest) and then push your changes back up to your repository, forcing an overwrite. Since you overwrote the server history too, we'll see only that new commit on GitHub. But you retained all the changes.

In your case, you want to do this

``` sh
$ git rebase origin/master
$ git rebase -i # choose to squash all but the first one; this command asks for each
$ git commit --amend # commit all the changes, adding them onto your existing commit
$ git push origin master -f # push it back, forcing the overwrite
```

For the above, use whatever ngboilerplate's remote is; I assumed it was origin, but you may have called it "ngBoilerplate". Try `git remote`.

PS: You should avoid working on the master branch as it makes change management difficult (for you). I like feature branches; so if I was going to develop this feature, I would check out  new branch from master:

``` sh
$ git checkout -b titleFromState # the '-b' means create the branch
```

And then I work off that branch, committing my changes as necessary. That way, I can work on other features on different branches without mixing everything together. For example, if you wanted to submit another pull request before this one got merged, it would be tough because your new PR would have these changes too. Also, if I was to recommend a small change in the PR, it would be more difficult for you to incorporate them cleanly in your existing codebase. Check out the Git book's chapter on [merging and branching](http://git-scm.com/book/en/Git-Branching-Basic-Branching-and-Merging) for some more technical know-how.

timkindberg commented:
Man I have just been struggling with this... thank you for your further explanation. I am going to attempt to follow your instructions. Sorry for the trouble.

timkindberg commented:
I was getting things to work it seemed, I had rebase working and was setting which commits to squash and which to pick. I think the real trouble here is that I didn't start on a new branch. So I'm just gonna delete the whole repo, refork and just put in all my changes in one commit. I know its lame, but that's where I'm at.

joshdmiller commented:
@ProLoser - I just had a chance to look at Pulley. I'll try it out with one of these PRs; your experience has been good?

ProLoser commented:
My experience has been non-existent, I just want to try adopting it.

On Fri, Jun 28, 2013 at 12:04 AM, Josh David Miller <
notifications@github.com> wrote:

> @ProLoser https://github.com/ProLoser - I just had a chance to look at
> Pulley. I'll try it out with one of these PRs; your experience has been
> good?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/joshdmiller/ng-boilerplate/pull/56#issuecomment-20173474
> .

## 

Dean J Sofer
DeanSofer@gmail.com

BS Computer Information Systems
California State Polytechnic University, Pomona
Telephone: 714-900-2254
Website: www.DeanSofer.com
Github: github.com/ProLoser

joshdmiller commented:
Landed as 33de809.

Many thanks, @timkindberg!!!

timkindberg commented:
Cool!
